File: contest/3/main.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {

    int t;
    cin >> t;

    for (int i = 0; i < t; ++i) {

        string s;
        cin >> s;

        for (int i = 1; i < s.size() - 1; ++i) {
            if (s[i - 1] != s[i] && s[i] != s[i + 1])
                s = s.substr(0, i) + s.substr(i + 1);
        }

        bool flag = true;
        for (int i = 0; i < s.size() - 1; ++i) {
            if (s[i] != s[i + 1]) {
                flag = false;
                break;
            }
        }

        if (flag) cout << "YES" << endl;
        else cout << "NO" << endl;
        
    }

    return 0;
    
}

File: contest/3/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	in := bufio.NewReader(os.Stdin)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()

	var t int
	fmt.Fscan(in, &t)

	for i := 0; i < t; i++ {
		var s string
		fmt.Fscan(in, &s)
		r := []rune(s)

		for i := 1; i < len(r)-1; i++ {
			if r[i-1] != r[i] && r[i] != r[i+1] {
				r = append(r[:i], r[i+1:]...)
			}
		}

		flag := true
		for i := 0; i < len(r)-1; i++ {
			if r[i] != r[i+1] {
				flag = false
				break
			}
		}

		if flag {
			fmt.Fprintln(out, "YES")
		} else {
			fmt.Fprintln(out, "NO")
		}
	}
}
